### PR TITLE
Internal API - CommandLineOptions for Project

### PR DESCRIPTION
**Pull request by TIHan**
This adds an *internal* property, `CommandLineOptions` to the workspace `Project` type. From the project shims, it can be set via `projectContext.SetOptions("this is command line args")` if the project context's language doesn't implement `CompilationOptions`; it will fallback and set the `CommandLineOptions` instead.

This is useful for command line arguments for F#, in particular, to flow through the workspace Project without having to be concerned with `CompilationOptions`.

### CODE REVIEW

File: Roslyn.sln [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.27110.0
+# Visual Studio Version 16
+VisualStudioVersion = 16.0.28417.3003
```
Comment (jmarolf) [inline Roslyn.sln]: Revert this change please

File: src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs [hunk 2]
```diff
@@ -239,6 +241,27 @@ public ParseOptions ParseOptions
...
+        {
+            get => _commandLineOptions;
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+
+                if (_activeBatchScopes > 0)
```
Comment (jmarolf) [inline src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs]: Shouldn't;t this be `_activeBatchScopes == 0`?

Comment (jmarolf) [reply src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs]: Actually doesn't `ChangeProjectProperty`  handle the batch scope check?

---
In reply to: [252487160](https://github.com/dotnet/roslyn/pull/32977#discussion_r252487160) [](ancestors = 252487160)

Comment (TIHan) [reply src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs]: Ah it actually does.

**Review by jmarolf — APPROVED** [review]

**Comment by jinujoseph** [comment]
@TIHan Closing this PR for now, feel free to update when you get an opportunity to work on this.
